src/service/resources-impl.js

```diff
@@ -932,7 +941,7 @@ export class Resources {
     for (let i = 0; i < this.resources_.length; i++) {
       const r = this.resources_[i];
```

Comment:
Filed an issue to track some refactoring here: #4450 - don't wanna get this PR too complicated - wanna check it in for the upcoming canary.